lib/ansiblelint/utils.py

```diff
@@ -365,18 +365,31 @@ def _kv_to_dict(v):
 def normalize_task_v2(task):
     """Ensure tasks have an action key and strings are converted to python objects."""
     result = dict()
-    mod_arg_parser = ModuleArgsParser(task)
+    clean_task = task.copy()
+    # task is an AnsibleMapping which inherits from OrderedDict, so we need
+    # to use `del` to remove unwanted keys.
+    for k in ['skipped_rules', FILENAME_KEY, LINE_NUMBER_KEY]:
+        if k in clean_task:
+            del clean_task[k]
+    if 'always_run' in task:
+        # FIXME(ssbarnea): Delayed import to avoid circular import
+        # See https://github.com/ansible/ansible-lint/issues/880
+        from ansiblelint.rules.AlwaysRunRule import AlwaysRunRule
+
+        raise MatchError(
+            rule=AlwaysRunRule,
+            filename=task[FILENAME_KEY],
+            linenumber=task[LINE_NUMBER_KEY])
+    mod_arg_parser = ModuleArgsParser(clean_task)
     try:
         action, arguments, result['delegate_to'] = mod_arg_parser.parse()
     except AnsibleParserError as e:
         try:
             task_info = "%s:%s" % (task[FILENAME_KEY], task[LINE_NUMBER_KEY])
-            del task[FILENAME_KEY]
-            del task[LINE_NUMBER_KEY]
         except KeyError:
             task_info = "Unknown"
         pp = pprint.PrettyPrinter(indent=2)
-        task_pprint = pp.pformat(task)
+        task_pprint = pp.pformat(clean_task)
```

Comment:
Why should it be clean?

Reply:
Ansible 2.10 no longer accepts extra/unknown arguments, like the ones we where using for filename and line number. We need to be sure we remove them when loading the task with ansible. Older versions did allow them.